core/src/main/java/ai/timefold/solver/core/impl/localsearch/DefaultLocalSearchPhase.java

```diff
@@ -130,7 +130,9 @@ public void stepStarted(LocalSearchStepScope<Solution_> stepScope) {
     public void stepEnded(LocalSearchStepScope<Solution_> stepScope) {
         super.stepEnded(stepScope);
         decider.stepEnded(stepScope);
-        collectMetrics(stepScope);
+        if (stepScope.isPhaseEnableCollectMetrics()) {
+            collectMetrics(stepScope);
+        }
```

Comment:
Why would metrics collection ever be disabled in Local Search? What changed that the metric collection is now conditional?

Also, let's change the method. Arguably, this belongs on the phase scope, and then it can be called `isMetricCollectionEnabled()`. If we need it, that is.

Reply:
> Why would metrics collection ever be disabled in Local Search? What changed that the metric collection is now conditional?

Do we want to account for the number of step moves in the R&R construction heuristic phase? If so, then we can remove the if block. However, I used the same idea from the move speed, viewing the R&R as a single move.

Reply:
The RnR should be viewed as a single move, that is definitely correct. But the RnR doesn't run Local Search, so why would this be affected? LS is _outside_ of the RnR move, it reports to the CH, so I'm not sure why those steps are reported here.

Reply:
Good point; they are not reported on the LS. Let me remove the if block.